core/embed/trezorhal/touch_t.h

```diff
@@ -115,6 +111,81 @@ void touch_power_on(void) {
     ensure(secfalse, NULL);
     return;
   }
+}
+
+void _i2c_deinit(void) {
+  if (i2c_handle.Instance) {
+    HAL_I2C_DeInit(&i2c_handle);
+    i2c_handle.Instance = NULL;
+  }
+}
+
+static void _i2c_ensure_pin(uint16_t GPIO_Pin, GPIO_PinState PinState) {
+  HAL_GPIO_WritePin(GPIOB, GPIO_Pin, PinState);
+  while (HAL_GPIO_ReadPin(GPIOB, GPIO_Pin) != PinState)
+    ;
+}
+
+void _i2c_cycle(void) {
+  _i2c_deinit();
+
+  // set above pins to OUTPUT / NOPULL
+  GPIO_InitTypeDef GPIO_InitStructure;
+
+  mp_printf(&mp_plat_print, "reset GPIOs...\n");
+  GPIO_InitStructure.Mode = GPIO_MODE_OUTPUT_PP;
```

Comment:
Errata says you should set Output Open Drain here, therefore `GPIO_MODE_OUTPUT_OD` is the correct constant.